src/D2L.CodeStyle.Analyzers/Extensions/Microsoft.CodeAnalysis.cs

```diff
@@ -104,7 +109,8 @@ internal static partial class RoslynExtensions {
 				}
 
 				var arg = attr.ConstructorArguments[0];
-				if( arg.Value.Equals( type ) ) {
+				//Using ToString, otherwise it sometimes fails to match, and the test behaviour does not match the real behaviour
+				if( arg.Value.ToString().Equals( type.ToString() ) ) {

```

Comment:
Not ideal, but without this I couldn't get the `ImplementsImmutableGeneric` test to work.  The ToString output was the same for the attribute and interface `ITypeSymbol`s ("System.IComparable<SpecTests.GenericsTests.ILocallyDefined>") but this line was returning false.

Reply:
What's the actual type of `Value` for that test? Trying to understand this.

Reply:
When the test is running, it looks like `type` is of type SourceNamedTypeSymbol, and `arg.Value` is of type ConstructedNamedTypeSymbol

Reply:
When running in the real DLL, both are of type ConstructedNamedTypeSymbol

Reply:
Hmmm....